core/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java

```diff
@@ -308,16 +338,26 @@ Query rangeQuery(String field, Object lowerTerm, Object upperTerm,
                 }
                 return fields;
             }
+
+            private void validateParsed(Float value) {
```

Comment:
please make it a `float` rather than `Float` and use `Float.isFinite(value) == false` below